### PR TITLE
Authentication Setup

### LINKED ISSUE #3: Authentication Setup
- Create user login and signup functionality
- Setup redux store and configure reducer to store user data
- Setup Login using firebase
- Store user data in a table in firestore

### PR DESCRIPTION
**Pull request by ttahm3d**
- Setup user signin and signout
- Added Google Auth Provider
- Store user data
- Private routes

Closes #3

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,71 +1,72 @@
...
-      {props.children}
+      {children}
     </StyledButton>
   );
 };
 
-const IconButton = (props: IconButtonProps) => {
+const IconButton = (props: IIconButtonProps) => {
   const { icon, ...rest } = props;
   return <StyledIconButton {...rest}>{props.icon}</StyledIconButton>;
```
Comment (kjais1720) [inline src/components/Button/Button.tsx:37]: you've already destructured the icon, so no need to write `props.icon`

File: src/components/ScrollToTop/ScrollToTop.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+interface IProps {
+  children?: JSX.Element | string;
+}
+
+export default function ScrollToTop({ children }: IProps): JSX.Element {
+  const location = useLocation();
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [location]);
+
```
Comment (kjais1720) [inline src/components/ScrollToTop/ScrollToTop.tsx:13]: If it's used on timeline the isn't it bad for UX that when user clicks on a post and then comes back to the timeline, they will be scrolled to the top and will have to scroll back to the previous position

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (ttahm3d) [reply src/components/Button/Button.tsx:37]: It was giving typescript error.

File: src/components/ScrollToTop/ScrollToTop.tsx (hunk 2 above)
Comment (ttahm3d) [reply src/components/ScrollToTop/ScrollToTop.tsx:13]: Havent come till this far to think about this, but will do when i come till there. For now keeping it.

**Review by vaibhav-mahalle — APPROVED** [review]
Liked the structured and well-thought-out commits, and also liked the commit messages.

**Pull request merged**